In [1]:
import numpy as np
import sys
import h5py
from utils.abundance_matching import AbundanceMatching
from utils.paths import SetupPaths
from utils.merger_trees import TraceMergerTree
from utils.get_groups import GetGroups

paths = SetupPaths()

imported h5py...
modules seem to have loaded?


In [2]:
snap = 99
sim = "TNG"

In [3]:
if sim == "Illustris":
    little_h = 0.704

elif sim == "TNG":
    little_h = 0.6774

kwargs = {"group_mass_min": 8,
          "group_mass_max": 50,
          "little_h": little_h}

In [4]:
groups = GetGroups(snapshot=snap,
                    sim=sim,
                    physics="hydro",
                    **kwargs)

In [5]:
groups.subhalo_masses

array([35.25347353, 30.68568329, 17.65063212, ...,  7.77373667,
        7.57665454,  7.51397516])

In [6]:
groups.subhalo_ids

array([471261, 471262, 471263, ..., 716606, 717038, 718824])

In [7]:
idcheck = groups.subhalo_ids[0]

In [8]:
max_mass_info = TraceMergerTree(snapshot=snap,
                subfindID=idcheck,
                sim="TNG", 
                physics="hydro",
                **kwargs)

In [9]:
max_mass_info.maxmass

(36.712093, 98)

In [10]:
max_mass_info.masses_phys

array([3.52534752e+01, 3.67120934e+01, 3.24774361e+01, 3.03779068e+01,
       1.50966530e+01, 2.72642040e+01, 2.52766037e+01, 2.41951256e+01,
       2.31076736e+01, 2.29044037e+01, 1.47792377e+01, 1.51824274e+01,
       1.44916992e+01, 1.47520905e+01, 1.52643442e+01, 1.46459513e+01,
       1.38758869e+01, 1.28782148e+01, 1.21509037e+01, 1.07517204e+01,
       9.96785736e+00, 1.05125732e+01, 1.09055576e+01, 1.05612154e+01,
       1.05222521e+01, 1.11451292e+01, 1.08712473e+01, 1.12110682e+01,
       1.12744131e+01, 1.16028175e+01, 1.20083637e+01, 1.20059347e+01,
       1.35281982e+01, 1.37735405e+01, 1.36350298e+01, 1.39942961e+01,
       1.41282244e+01, 1.39739628e+01, 1.37122936e+01, 1.34031181e+01,
       1.31577854e+01, 1.29171009e+01, 1.25806580e+01, 1.22211761e+01,
       1.21676197e+01, 1.16914883e+01, 1.11645670e+01, 1.05225344e+01,
       1.00021639e+01, 9.35938263e+00, 9.13931179e+00, 8.80828953e+00,
       8.49601746e+00, 8.28941250e+00, 8.10124493e+00, 7.90900087e+00,
      

In [11]:
max_mass_info.masses

array([2.3880703e+01, 2.4868773e+01, 2.2000216e+01, 2.0577993e+01,
       1.0226473e+01, 1.8468771e+01, 1.7122372e+01, 1.6389778e+01,
       1.5653138e+01, 1.5515443e+01, 1.0011456e+01, 1.0284576e+01,
       9.8166771e+00, 9.9930658e+00, 1.0340067e+01, 9.9211674e+00,
       9.3995256e+00, 8.7237024e+00, 8.2310219e+00, 7.2832155e+00,
       6.7522264e+00, 7.1212168e+00, 7.3874249e+00, 7.1541672e+00,
       7.1277733e+00, 7.5497103e+00, 7.3641829e+00, 7.5943775e+00,
       7.6372876e+00, 7.8597484e+00, 8.1344652e+00, 8.1328201e+00,
       9.1640015e+00, 9.3301964e+00, 9.2363691e+00, 9.4797363e+00,
       9.5704594e+00, 9.4659624e+00, 9.2887077e+00, 9.0792723e+00,
       8.9130840e+00, 8.7500439e+00, 8.5221376e+00, 8.2786245e+00,
       8.2423458e+00, 7.9198141e+00, 7.5628777e+00, 7.1279650e+00,
       6.7754660e+00, 6.3400455e+00, 6.1909695e+00, 5.9667354e+00,
       5.7552023e+00, 5.6152482e+00, 5.4877830e+00, 5.3575573e+00,
       5.3124843e+00, 5.3637252e+00, 5.5373855e+00, 5.3339891e

In [12]:
np.max(max_mass_info.masses_phys)

36.712093

In [13]:
path = f"TNG_hydro_99.hdf5"
data = h5py.File(f"{paths.path_maxmass}{path}", "r")

In [14]:
dictt = {}
for key in data.keys():
    dictt[key] = np.array(data[key])

In [15]:
dictt

{'Current Snap Mass': array([304.52587891, 107.17776489,  37.22546387, ...,   5.26592922,
          5.13242579,   5.08996677]),
 'Group ID': array([  216,   216,   216, ..., 24272, 24384, 24833], dtype=int32),
 'Max Mass': array([452.28439331, 188.17633057,  68.26158142, ...,   7.81411791,
          7.6967454 ,  11.45020485]),
 'Max Mass Snap': array([90, 95, 90, ..., 90, 93, 51]),
 'Subhalo ID': array([327822, 327823, 327824, ..., 716606, 717038, 718824])}

In [16]:
mask = dictt['Subhalo ID'] == idcheck

In [17]:
dictt["Max Mass"][mask]

array([36.71209335])

In [18]:
dictt["Current Snap Mass"][mask]

array([23.88070297])

In [19]:
groups.subhalo_masses[0]

35.25347353470934

In [20]:
dictt["Current Snap Mass"][mask] / little_h

array([35.25347353])

Looks like the Current snap mass stored in the max mass catalogs are incorrect! (should be 35.25) # need to divide current snap mass in max mass catalog by little h

In [24]:
ampath = f"TNG_hydro_99.hdf5"
amdata = h5py.File(f"{paths.path_am_mass}{ampath}", "r")

In [25]:
amdata.keys()

<KeysViewHDF5 ['Current Snap Mass', 'Group ID', 'Max Mass', 'Max Mass Snap', 'Median Stellar Mass', 'Stellar Masses', 'Subhalo ID']>

In [26]:
amdictt = {}
for key in amdata.keys():
    amdictt[key] = np.array(amdata[key])

In [27]:
ammask = amdictt['Subhalo ID'] == idcheck

In [28]:
for key in amdata.keys():
    print(key, amdictt[key][ammask])

Current Snap Mass [23.88070297]
Group ID [1245]
Max Mass [36.71209335]
Max Mass Snap [98]
Median Stellar Mass [1.25812685]
Stellar Masses [[1.07934352 0.68909576 0.90780967 0.83024818 1.18052597 1.01421812
  0.65266029 0.88860345 1.27045187 1.27485301 1.12176586 1.25324321
  0.52918981 1.10841055 0.92050031 0.89971668 1.04999801 1.04339
  1.15178785 1.28015093 1.14322339 1.10634598 1.42811332 0.79924
  1.13935557 0.53311907 1.00919021 1.2972873  1.52466582 1.7913591
  1.24712229 0.75433894 1.12357813 0.85585464 1.04601956 1.6046601
  0.548823   0.93248398 1.28224827 1.41165267 1.34846731 1.26283438
  1.24248964 1.40449368 1.43376731 1.19398353 1.47493122 1.16174656
  0.82004317 1.47680917 0.78263206 0.76347326 1.07358567 1.3968511
  1.22313788 1.07990293 1.56068707 1.4389643  0.81622283 1.28160206
  1.23181959 0.93645321 1.25031843 0.78613781 0.93373856 0.38113847
  1.31474657 0.99947408 0.80283522 0.96933813 1.30099729 1.07850028
  1.2906983  0.85304164 1.2505648  1.01621241 1.5608295

current mass is also incorrect in the abundance matching data

need to make sure than cut on halo mass of 1e9 is physical not Msun/h! 

In [29]:
amdata.close()
data.close()